### PR TITLE
set px version in status during migration

### PR DESCRIPTION
**Pull request by haibinxie**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/migration/generate.go [hunk 1]
```diff
@@ -835,6 +835,7 @@ func (h *Handler) handleCustomImageRegistry(cluster *corev1.StorageCluster) erro
 		MetricsCollectorProxy:      h.removeCustomImageRegistry(cluster.Spec.CustomImageRegistry, cluster.Status.DesiredImages.MetricsCollectorProxy),
 		PxRepo:                     h.removeCustomImageRegistry(cluster.Spec.CustomImageRegistry, cluster.Status.DesiredImages.PxRepo),
 	}
+	cluster.Status.Version = pxutil.GetImageTag(cluster.Spec.Image)
```
Comment (piyush-nimbalkar) [inline pkg/migration/generate.go:838]: Why do we need this? This will be automatically set by the operator when the migration starts.

Comment (haibinxie) [reply pkg/migration/generate.go:838]: If we dont have this, the desiredImages will be overwritten after stc reconcile.

Comment (piyush-nimbalkar) [reply pkg/migration/generate.go:838]: Oh yes, that would happen.

Comment (haibinxie) [reply pkg/migration/generate.go:838]: Good point, changed

**Review by piyush-nimbalkar — APPROVED** [review]
Let's add this check in the existing unit test as well.

**Comment by haibinxie** [comment]
> 

I was trying to change unit test but turned out complicated to mock controller reconcile in migration unit tests. I will add functional tests in next sprint which would cover this as well, in next sprint I will list out the plan for all functional tests.

**Pull request merged**